wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt

```diff
@@ -1022,20 +1022,22 @@ class KotlinGenerator private constructor(
   ): List<Pair<ParameterSpec, PropertySpec>> {
     val result = mutableListOf<Pair<ParameterSpec, PropertySpec>>()
 
+    var index = 0
     for (fieldOrOneOf in message.fieldsAndFlatOneOfFieldsAndBoxedOneOfs()) {
       when (fieldOrOneOf) {
         is Field -> result.add(
           constructorParameterAndProperty(
-            message,
-            fieldOrOneOf,
-            nameAllocator
+            message = message,
+            field = fieldOrOneOf,
+            nameAllocator = nameAllocator,
+            constructorParameterIndex = index++,
```

Comment:
Does `message.fieldsAndFlatOneOfFieldsAndBoxedOneOfs()` return the fields in the order they're declared in the constructor? Or maybe I'm misunderstanding the approach?

Reply:
No, your understanding is correct.